### PR TITLE
Drop constant and payable from ABI.

### PR DESCRIPTION
**Pull request by ekpyron**
Fixes https://github.com/ethereum/solidity/issues/4755

### CODE REVIEW

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Conflict.

**Comment by ekpyron** [comment]
Updated. I wonder whether it's possible to avoid Changelog merge conflicts by adding some git attribute magic...

File: docs/abi-spec.rst [hunk 1]
```diff
@@ -454,16 +454,11 @@ A function description is a JSON object with the fields:
 
 - ``outputs``: an array of objects similar to ``inputs``, can be omitted if function doesn't return anything;
 - ``stateMutability``: a string with one of the following values: ``pure`` (:ref:`specified to not read blockchain state <pure-functions>`), ``view`` (:ref:`specified to not modify the blockchain state <view-functions>`), ``nonpayable`` (function does not accept Ether) and ``payable`` (function accepts Ether);
-- ``payable``: ``true`` if function accepts Ether, ``false`` otherwise;
-- ``constant``: ``true`` if function is either ``pure`` or ``view``, ``false`` otherwise.
 
-``type`` can be omitted, defaulting to ``"function"``, likewise ``payable`` and ``constant`` can be omitted, both defaulting to ``false``.
+``type`` can be omitted, defaulting to ``"function"``.
```
Comment (axic) [inline docs/abi-spec.rst]: Can we perhaps remove this quirk too? :)

Comment (ekpyron) [reply docs/abi-spec.rst]: Added another commit removing it.

**Review by chriseth — APPROVED** [review]

**Pull request merged**